### DISCUSSION TITLE
Vdostream speed question

### DISCUSSION THREAD

**Discussion by Cacsjep** [AxisCommunications/acap3-examples#187] [Q&A] [answered]
**Hey Guys,**

When getting frame and buffer from the vdo_stream it takes on my test camera around 40ms, currently I just testing with the vdoencodeclient example, and do this:
```
gettimeofday(&start, NULL);
VdoBuffer* buffer = vdo_stream_get_buffer(stream, &error);
VdoFrame*  frame  = vdo_buffer_get_frame(buffer);
if (!frame)
    goto exit;
gettimeofday(&stop, NULL);
printf("took %lu ms\n", ((stop.tv_sec - start.tv_sec) * 1000000 + stop.tv_usec - start.tv_usec)/ 1000); 
```
Test Stream:  JPEG 30FPS 640x360 

I am wondering if there an way to speed this up or if this already the fastest way or limit, thank u.

Br christoph

**Comment by pataxis** [comment]
Hi @Cacsjep , thanks for your question.
What device and firmware version are you using?

**Comment by Cacsjep** [comment, marked as answer]
Hi @pataxis,

Thank u for your response,
I am now testing on a AXIS Q1715 Block Camera with firmware 11.0.93

What I notice is that when I change the appearance compression to 0, FULLHD goes down to 0ms other resolutions are around 40ms but slighty decrased in average 33-38ms.
![image](https://user-images.githubusercontent.com/98898150/207289590-a9cd2f80-7673-4b71-a364-8c26fca625ec.png)

**Reply by daniel-nielsen-se** [reply]
Hi Cacsjep,

there is currently no way to speed up the first frame, but the delta of subsequent frames should be 1/30s. Depending on what you are trying to accomplish, you could leave the stream running permanently and only call get_buffer when you actually need the frame.

Best Regards,
Daniel N

**Reply by Cacsjep** [reply]
OK sounds good to me I test it with an Q1615 with 100Fps and I am know having 10ms, its not about the first frame, we look into deep how far we come to reduce the overall latency. Thank u, that info helps me
